app/helpers/formatPapersQuery.tsx

```diff
@@ -0,0 +1,47 @@
+interface IFormatPapersQueryParams {
+  text: string;
+  yearFrom?: number;
+  yearTo?: number;
+}
+
+export function formatPapersQuery({ text, yearFrom, yearTo }: IFormatPapersQueryParams) {
+  const resultQuery = `text=${text},year=${yearFrom || ""}:${yearTo || ""}`;
+
+  return encodeURIComponent(resultQuery);
+}
+
+export function objectifyPapersQuery(query: string): { text: string; yearFrom?: number; yearTo?: number } {
+  const textIndex = query.search("text=");
+  const yearFromIndex = query.search("year=");
+  const yearToIndex = query.search(":");
```

Comment:
How to get this value when there is additional field with colon?

Reply:
I will fix it.